packages/salesforcedx-vscode-apex-replay-debugger/src/index.ts

```diff
@@ -44,24 +46,53 @@ function registerCommands(): vscode.Disposable {
         defaultUri: getDialogStartingPath()
       });
       if (fileUris && fileUris.length === 1) {
-        lastOpenedLogFolder = path.dirname(fileUris[0].fsPath);
+        extContext.workspaceState.update(
```

Comment:
Wow, I haven't used this API so I could be wrong. But these calls are synchronous (instead of asynchronous and returning a promise)? 

Interesting. The updates to the keystore must be pretty fast then.